### PR TITLE
Switch to http-call and implement retries

### PR DESCRIPTION
**Pull request by dmathieu**
I will be moving this to `heroku-cli-util` to other plugins can use it. Doing it here first so we can ship it.

### CODE REVIEW

File: lib/streamer.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+const http = require('http-call').HTTP
+const maxRetries = 10
+
+async function call (url, out, retries) {
```
Comment (jdx) [inline lib/streamer.js:4]: If you made this `call (url, out, retries = 0)` you won't need the extra function

**Review by jdx — CHANGES_REQUESTED** [review]
Remove the `package-lock.json` and use `yarn`

File: lib/streamer.js (hunk 1 above)
Comment (dmathieu) [reply lib/streamer.js:4]: The method signatures are different. We don't allow setting the `retries` argument in the exported method.

**Comment by dmathieu** [comment]
@jdxcode thanks. I've used yarn instead.

**Review by jdx — APPROVED** [review]

**Review by ojacobson — APPROVED** [review]
Looks reasonable to me. I had a read through the `streamer` implementation and I think it's correct. The tests appear to corroborate that.

**Pull request merged**